cns/restserver/ipam.go

```diff
@@ -478,101 +634,194 @@ func (service *HTTPRestService) MarkExistingIPsAsPendingRelease(pendingIPIDs []s
 	return nil
 }
 
-func (service *HTTPRestService) GetExistingIPConfig(podInfo cns.PodInfo) (cns.PodIpInfo, bool, error) {
-	var (
-		podIpInfo cns.PodIpInfo
-		isExist   bool
-	)
+func (service *HTTPRestService) GetExistingIPConfig(podInfo cns.PodInfo) ([]cns.PodIpInfo, bool, error) {
+	podIPInfo := make([]cns.PodIpInfo, len(service.PodIPIDByPodInterfaceKey[podInfo.Key()]))
 
 	service.RLock()
 	defer service.RUnlock()
 
-	ipID := service.PodIPIDByPodInterfaceKey[podInfo.Key()]
-	if ipID != "" {
-		if ipState, isExist := service.PodIPConfigState[ipID]; isExist {
-			err := service.populateIPConfigInfoUntransacted(ipState, &podIpInfo)
-			return podIpInfo, isExist, err
+	for i, ipID := range service.PodIPIDByPodInterfaceKey[podInfo.Key()] {
+		if ipID != "" {
+			if ipState, isExist := service.PodIPConfigState[ipID]; isExist {
+				err := service.populateIPConfigInfoUntransacted(ipState, &podIPInfo[i])
+				if i == len(service.PodIPIDByPodInterfaceKey[podInfo.Key()])-1 {
```

Comment:
instead of returning with this if, I would suggest, keep count of numIPConfigsFound and if those match expectedNumIpConfigs = len(service.PodIPIDByPodInterfaceKey[podInfo.Key()]) then return success.

Reply:
I agree that will look cleaner